### ISSUE TITLE
EntityDataRepository::get() inserts a stub for unknown entity type IDs, which leaks into the entity-type-id type

### ISSUE THREAD

**Issue by mglaman** [mglaman/phpstan-drupal#1054]
## Problem

`EntityDataRepository::get()` inserts a stub `EntityData` for any entity type ID it has not seen:

```php
public function get(string $entityTypeId): EntityData
{
    if (!isset($this->entityData[$entityTypeId])) {
        $this->entityData[$entityTypeId] = new EntityData($entityTypeId, []);
    }
    return $this->entityData[$entityTypeId];
}
```

`getAllEntityTypeIds()` returns `array_keys($this->entityData)`, so every stub becomes a "known" entity type ID. `EntityTypeIdTypeNodeResolverExtension` feeds that list unfiltered into `EntityTypeIdType`, which uses it to decide whether a constant string is a valid entity type ID.

## Reproduction

Two files analyzed by the same worker, in this order:

```php
// a.php
$entityRepository->getActive('nodee', 5);
```

```php
// b.php
/** @param entity-type-id $id */
function foo(string $id): void {}
foo('nodee');
```

The `getActive()` call in `a.php` inserts a stub for `nodee`. The call in `b.php` is then accepted because `nodee` is now in the entity type ID list. Analyze `b.php` alone and it is reported. The result depends on file analysis order across workers.

## Affected callers

Every return type extension that looks up a constant string calls `get()`:

- `DrupalStaticEntityQueryDynamicReturnTypeExtension`
- `EntityRepositoryReturnTypeExtension`
- `EntityStorage\EntityStorageDynamicReturnTypeExtension`
- `EntityTypeManagerGetStorageDynamicReturnTypeExtension`

## Fix

Make the lookup non-mutating. Two options:

1. Add `has(string $entityTypeId): bool` and have callers check it before `get()`, or return a throwaway `EntityData` from `get()` without storing it.
2. Keep the stub cache but have `getAllEntityTypeIds()` return only IDs that came from the configured entity map.

Option 1 is cleaner because it removes the shared mutable state entirely. Either way, add a test that resolves an unknown ID through a return type extension and then asserts `getAllEntityTypeIds()` does not contain it.

Found while reviewing #1041.
